org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/analysis/SliceDeviceContext.java

```diff
@@ -0,0 +1,90 @@
+package org.eclipse.scanning.sequencer.analysis;
+
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+
+public class SliceDeviceContext {
+
+	private IPosition    location;
+	private IScanSlice   scanSlice;
+	private ILazyDataset data;
+	private IDataset     slice;
+	
+	public SliceDeviceContext() {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1186)